### DISCUSSION TITLE
Ability to configure what `no_multiline_whitespace_around_double_arrow` fixes

### DISCUSSION THREAD

**Discussion by lstrojny** [PHP-CS-Fixer/PHP-CS-Fixer#7314] [Feature ideas]
`no_multiline_whitespace_around_double_arrow` is applied for arrow functions. That clashes with multi line formatting of arrow functions.

```php
$fn = fn(string $someRatherLongVariableName) =>
    $someRatherLongVariableName === $theOtherPrettyLongVariableNameOneMightWantToCompareThingsWith;
```
`no_multiline_whitespace_around_double_arrow` will reformat this in the following way and [apparently that is a conscious decision](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/6cee2bedbb432941a0c894ab35a9920b6ffe8562/tests/Fixer/ArrayNotation/NoMultilineWhitespaceAroundDoubleArrowFixerTest.php#L107):

```php
$fn = fn(string $someRatherLongVariableName) => $someRatherLongVariableName === $theOtherPrettyLongVariableNameOneMightWantToCompareThingsWith;
```

Wouldn't it make sense that the double arrow space formatting would either not apply to arrow functions at all or that it can be turned off and on for arrays or arrow functions?

**Comment by raymondschouten** [comment]
I'm running into this issue as well with `match` statements. To me making configurable in which context to disallow the multiline whitespace makes sense!
